include/glaze/ext/jsonrpc.hpp

```diff
@@ -0,0 +1,548 @@
+      {
+         using char_type = CharType;
+
+         char_type data_[N + 1]{};
+
+         constexpr basic_fixed_string() noexcept : data_{} {}
+
+         template <typename other_char_type>
+            requires std::same_as<other_char_type, char_type>
+         constexpr basic_fixed_string(const other_char_type (&foo)[N + 1]) noexcept
+         {
+            std::copy_n(foo, N + 1, data_);
+         }
+
+         [[nodiscard]] constexpr std::basic_string_view<char_type> view() const noexcept { return {&data_[0], N}; }
+
+         constexpr operator std::basic_string_view<char_type>() const noexcept { return {&data_[0], N}; }
+
+         template <unsigned M>
+         constexpr auto operator==(basic_fixed_string<char_type, M> const& r) const noexcept
+         {
+            return N == M && view() == r.view();
+         }
+      };
+
+      template <typename char_type, unsigned N>
+      basic_fixed_string(char_type const (&str)[N]) -> basic_fixed_string<char_type, N - 1>;
+
+      template <typename char_type, std::size_t N, std::size_t M>
+      consteval auto concat_fixed_string(basic_fixed_string<char_type, N> l,
+                                         basic_fixed_string<char_type, M> r) noexcept
+      {
+         basic_fixed_string<char_type, N + M> result;
+         auto it{std::copy(l.begin(), l.end(), result.begin())};
+         it = std::copy(r.begin(), r.end(), it);
+         *it = {};
+         return result;
+      }
+
+      template <typename char_type, std::size_t N, std::size_t M, typename... U>
+      consteval auto concat_fixed_string(basic_fixed_string<char_type, N> l, basic_fixed_string<char_type, M> r,
+                                         U... u) noexcept
+      {
+         return concat_fixed_string(l, concat_fixed_string(r, u...));
+      }
+   }
+
+   class error
+   {
+     public:
+      static error invalid(parse_error const& pe, auto& buffer)
+      {
+         std::string format_err{format_error(pe, buffer)};
+         return error(rpc::error_e::invalid_request, format_err.empty() ? glz::json_t{} : format_err);
+      }
+      static error version(std::string_view presumed_version)
+      {
+         return error(error_e::invalid_request, "Invalid version: " + std::string(presumed_version) +
+                                                   " only supported version is " + std::string(rpc::supported_version));
+      }
+      static error method(std::string_view presumed_method)
+      {
+         return error(error_e::method_not_found, "Method: \"" + std::string(presumed_method) + "\" not found");
+      }
+
+      error() = default;
+
+      explicit error(error_e code, glz::json_t&& data = {})
+         : code(util::to_underlying(code)), message(code_as_string(code)), data(std::move(data))
+      {}
+
+      [[nodiscard]] auto get_code() const noexcept -> std::optional<error_e>
+      {
+         auto const it{std::find_if(std::cbegin(error_e_iterable), std::cend(error_e_iterable),
```

Comment:
yes I was not sure how to implement this since I cannot know whether the parsed value is legal value, and if some user has used a value outside the specification he should not get a parsing error and be able to get the raw value.

Reply:
This was a note to myself that we should add this functionality to glaze for enums exposed to the meta system because it seems useful.